app/battle/creating_knights.py

```diff
@@ -0,0 +1,17 @@
+from app.knights.knights import Knight
+
+
+def creating_knight(name: str, knights_config: dict) -> Knight:
+    config = knights_config[name]
+    new_knight = Knight(
+        name=config["name"],
+        power=config["power"],
+        hp=config["hp"]
+    )
+    new_knight.apply_armour(armour=config["armour"])
+    new_knight.apply_weapon(weapon=config["weapon"])
+    new_knight.apply_potion(potion=config["potion"])
```

Comment:
The `apply_potion` method should handle cases where `potion` might be `None`. Ensure that the function does not attempt to apply effects if the potion is `None`.